homeassistant/components/input_boolean/__init__.py

```diff
@@ -27,23 +33,56 @@
 
 CONF_INITIAL = "initial"
 
+CREATE_FIELDS = {
+    vol.Required(CONF_NAME): vol.All(str, vol.Length(min=1)),
+    vol.Optional(CONF_INITIAL): cv.boolean,
+    vol.Optional(CONF_ICON): cv.icon,
+}
+
+UPDATE_FIELDS = {
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_INITIAL): cv.boolean,
+    vol.Optional(CONF_ICON): cv.icon,
+}
+
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: cv.schema_with_slug_keys(
-            vol.Any(
-                {
-                    vol.Optional(CONF_NAME): cv.string,
-                    vol.Optional(CONF_INITIAL): cv.boolean,
-                    vol.Optional(CONF_ICON): cv.icon,
-                },
-                None,
-            )
-        )
-    },
+    {DOMAIN: cv.schema_with_slug_keys(vol.Any(UPDATE_FIELDS, None))},
     extra=vol.ALLOW_EXTRA,
 )
 
 RELOAD_SERVICE_SCHEMA = vol.Schema({})
+STORAGE_KEY = DOMAIN
+STORAKE_VERSION = 1
+
+
+class InputBooleanStorageCollection(collection.StorageCollection):
+    """Input boolean collection stored in storage."""
+
+    CREATE_SCHEMA = vol.Schema(CREATE_FIELDS)
+    UPDATE_SCHEMA = vol.Schema(UPDATE_FIELDS)
+
+    async def _process_create_data(self, data: typing.Dict) -> typing.Dict:
+        """Validate the config is valid."""
+        return self.CREATE_SCHEMA(data)
+
+    @callback
+    def _get_suggested_id(self, info: typing.Dict) -> str:
+        """Suggest an ID based on the config."""
+        return info[CONF_NAME]
+
+    async def _update_data(self, data: dict, update_data: typing.Dict) -> typing.Dict:
+        """Return a new updated data object."""
+        return self.UPDATE_SCHEMA(update_data)
```

Comment:
This is not correct. `update_data` might not contain all the fields, it need to be merged with the old data.

Reply:
I see. will fix.